configure.ac

```diff
@@ -98,6 +98,17 @@ MODPLUG_LIBRARY_VERSION=1:0:0
 
 AC_SUBST(MODPLUG_LIBRARY_VERSION)
 
+case "${host_os}" in
+	cygwin*|mingw*)
+		LIBS_PRIVATE="-luser32"
+		;;
```

Comment:
This is wrong: mingw and cygwin are gcc-based and need -lstdc++ (same with the cmake side.)

Reply:
Good catch. What should be the correct host_os that corresponds to cmake's WIN32?

Reply:
I think cmake sets `MSVC` for msvc environment.

If it were me, I'd do something like the following for win32 case in cmake:
```
if(WIN32)
  set(LIBS_PRIVATE "-luser32")
  if(MINGW OR CYGWIN)
    set(LIBS_PRIVATE "${LIBS_PRIVATE}"-lstdc++")
  endif()
else()
  [....]
```

Reply:
Ok, perhaps we can assume that autotools won't be used with msvc (although it's possible with msys2). For cmake, I've updated the patch to check with MSVC. I'll see if I can run a test with mingw.